vendor/github.com/docker/libnetwork/agent.go

```diff
@@ -463,52 +463,64 @@ func (ep *endpoint) deleteDriverInfoFromCluster() error {
 }
 
 func (ep *endpoint) addServiceInfoToCluster() error {
+	if ep.isAnonymous() && len(ep.myAliases) == 0 || ep.Iface().Address() == nil {
```

Comment:
Yes, the aliases, which come from the create configuration via options, are populated once the endpoint object is created and never changed.